scrapers/nus-v2/src/tasks/GetSemesterTimetable.ts

```diff
@@ -111,7 +111,20 @@ export function transformModgrpToClassNo(modgrp: string, activity: string): stri
 }
 
 export function mapTimetableLesson(lesson: TimetableLesson, logger: Logger): TempRawLesson {
-  const { room, start_time, end_time, day, module, modgrp, activity, eventdate, csize } = lesson;
+  const {
+    room: providedRoom,
```

Comment:
This avoids the `|| ''` and `?.` below 

```suggestion
    room: providedRoom = '',
```